src/main/java/com/optiva/tools/addevents/NatsEventPublisher.java

```diff
@@ -50,33 +27,29 @@ public NatsEventPublisher(final NatsConfiguration configuration) {
      * @param subject String representation of message subject e.g. 100.23.66.66
      * @throws NatsEventException custom exception to hold any errors while publishing
      */
-    public void publish(ByteArrayOutputStream event, String subject) throws NatsEventException {
+    public void publish(ByteArrayOutputStream event, String subject, String streamName) throws NatsEventException {
         try {
-            Message msg = NatsMessage.builder().subject(subject).data(event.toByteArray()).build();
-            PublishAck ack = jetStream.publish(msg, publishOptions);
-            if (ack != null) {
+            CompletableFuture<PublishAck> futureAck = AsyncPublishingConnectionManager.getInstance().publishAysnc(event, subject, streamName);
+            if (futureAck != null) {
+                PublishAck ack = futureAck.get(30, TimeUnit.SECONDS);;
```

Comment:
this is too long for timeout. I think we need to fail faster. Whats your opinion. create new connection and retry after 1-2 seconds. If this could be configurable paramter in milli-seconds

Reply:
i agree its not easy to pick one so we might need to tweak this